### PR TITLE
gnrc_rpl_p2p: several compilation fixes

### LINKED ISSUE #13369: native:in file gnrc_rpl_p2p.c function trickle_start() has extra and some undeclared variables.
Screenshot from 2020-02-13 19-17-21

### PR DESCRIPTION
**Pull request by cgundogan**
### Contribution description
A couple of issues appear when compiling `gnrc_rpl_p2p` (reported in #13369).
This PR fixes the cleanup timer usage and uses the correct netif functions.

### Testing procedure
* add `USEMODULE += gnrc_rpl_p2p` to `examples/gnrc_networking`.
* start two or more native instances
* add a global address to one native instance `ifconfig 6 add abcd::1` (neighbour discovery propagates a prefix to the other native instances).
* start p2p-rpl on the node that configured `abcd::1` to find a path to one of the native instances (assuming a global address of `abcd::bc63:a1ff:fe23:df98`):
`rpl find abcd::1 abcd::bc63:a1ff:fe23:df98`
* `rpl show` lists a temporary dodag
* `nib route` should show a forwarding entry to the second native instance

### Issues/PRs references
fixes #13369

### CODE REVIEW

**Comment by cgundogan** [comment]
@avinigam does this PR solve the issue for you?

**Comment by cgundogan** [comment]
@miri64 thanks for the quick review. Is the provided compile tests sufficient enough?

File: tests/gnrc_rpl_p2p/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Copyright (C) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+from testrunner import run
+
+def testfunc(child):
```
Comment (miri64) [inline tests/gnrc_rpl_p2p/tests/01-run.py]: Extra newline required according to `flake8`.

Comment (cgundogan) [reply tests/gnrc_rpl_p2p/tests/01-run.py]: pushed a fixup

**Comment by miri64** [comment]
The test application also needs a `Makefile.ci`.

File: tests/gnrc_rpl_p2p/Makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+include ../Makefile.tests_common
+
+USEMODULE += gnrc_netdev_default
+USEMODULE += auto_init_gnrc_netif
```
Comment (miri64) [inline tests/gnrc_rpl_p2p/Makefile]: For `native` the application tries to access a TAP interface with `gnrc_netdev_default`. As this is just a compile test you can remove these two modules.

**Comment by cgundogan** [comment]
@miri64 somehow murdock complains about an unsuccessful micropython test .. seems completely unrelated to this PR, though ..

**Comment by miri64** [comment]
Yepp, that's one of our regular false positives :-/

**Pull request merged**